interactions/api/models/guild.py

```diff
@@ -236,6 +239,144 @@ def __init__(self, **kwargs):
             else None
         )
 
+    async def ban(
+        self,
+        member_id: int,
+        reason: Optional[str] = None,
+        delete_message_days: Optional[int] = 0,
+    ) -> None:
+        """
+        Bans a member from the guild
+        :param member_id: The id of the member to ban
+        :type member_id: int
+        :param reason?: The reason of the ban
+        :type reason: Optional[str]
+        :param delete_message_days?: Number of days to delete messages, from 0 to 7. Defaults to 0
+        :type delete_message_days: Optional[int]
+        """
+        await self.client.create_guild_ban(
+            guild_id=int(self.id),
+            user_id=member_id,
+            reason=reason,
+            delete_message_days=delete_message_days,
+        )
+
+    async def remove_ban(
+        self,
+        user_id: int,
+        reason: Optional[str] = None,
+    ) -> None:
+        """
+        Removes the ban of a user
+        :param user_id: The id of the user to remove the ban from
+        :type user_id: int
+        :param reason?: The reason for the removal of the ban
+        :type reason: Optional[str]
+        """
+        await self.client.remove_guild_ban(
+            guild_id=int(self.id),
+            user_id=user_id,
+            reason=reason,
+        )
+
+    async def kick(
+        self,
+        member_id: int,
+        reason: Optional[str] = None,
+    ) -> None:
+        """
+        Kicks a member from the guild
+        :param member_id: The id of the member to kick
+        :type member_id: int
+        :param reason?: The reason for the kick
+        :type reason: Optional[str]
+        """
+        await self.client.create_guild_kick(
+            guild_id=int(self.id),
+            user_id=member_id,
+            reason=reason,
+        )
+
+    async def add_member_roles(
+        self,
+        roles: Union[List[Union[Role, int]], Role, int],
```

Comment:
This type hint doesn't respect the current pattern that we have for singulars and multiples. Please consider `Union[List[Role], Role, int]` instead.